### PR TITLE
feat: add facet-based config engine crates

### PR DESCRIPTION
**Pull request by jarhodes314**
## Proposed changes
Port the reflection-based configuration engine from the facet-config prototype. The engine provides generic config I/O (defaults, readers, schema reflection, federated mounts) without any tedge-specific types — those remain in tedge_config/models.

These changes comprise the groundwork for integrating the `mappers.<name>.<key>` syntax into tedge config, which will be added by a follow-up PR (to be later followed by replacing the entire `tedge_config` definition with the new reflection-based approach). You can treat the crates added in this PR as a replacement for `tedge_config_macros` and `tedge_config` (minus the model definitions which will remain even with the facet migration, and the actual config definition using the macro).

The PR adds three crates:
- tedge_config_engine: runtime support (reflect, defaults, reader, ops, append-remove, federated config)
- tedge_config_engine-macro: proc macro exposing define_config!
- tedge_config_engine-impl: macro implementation with unit tests

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s. You can activate automatic signing by running `just prepare-dev` once)
- [x] I ran `just format` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `just check` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/common/tedge_config_engine/src/append_remove.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+impl_single_value!(
+    String,
+    u16,
+    u32,
+    u64,
+    std::net::IpAddr,
+    std::num::NonZeroU16,
+    camino::Utf8PathBuf,
+    bool,
+);
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/src/append_remove.rs:72]: Support for collection types like `TemplateSet` were missed or to be added later?

File: crates/common/tedge_config_engine/macro/tests/compile_fail/derive_fn_returns_wrong_type.stderr [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  --> tests/compile_fail/derive_fn_returns_wrong_type.rs:14:79
+   |
+14 |             #[tedge_config(default(from_key_via(key = "mqtt.port", function = "next_port")))]
+   |                                                                               ^^^^^^^^^^^
+   |                                                                               |
+   |                                                                               expected fn pointer, found fn item
+   |                                                                               arguments to this function are incorrect
+   |
+   = note: expected fn pointer `for<'a> fn(&'a str) -> std::result::Result<std::option::Option<u16>, std::string::String>`
+                 found fn item `for<'a> fn(&'a str) -> std::result::Result<std::option::Option<std::string::String>, std::string::String> {next_port}`
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/macro/tests/compile_fail/derive_fn_returns_wrong_type.stderr:11]: Matching the errors on this entire error message which is heavily formatted seems a bit fragile, although I agree that the compiler changing its error reporting style is going to be a rare event. But still, to be on the safer side, is it possible to assert only on this relevant part of the error message?

File: crates/common/tedge_config_engine/src/defaults.rs [hunk 3]
```diff
@@ -0,0 +1,726 @@
...
+use std::borrow::Cow;
+
+/// Describes how to produce a value when a config key is not set directly.
+#[derive(Debug)]
+pub enum DefaultSpec {
+    /// A fixed default value stored as the same string accepted by `config set`.
+    Value(String),
+
+    /// A fresh default computed each time the key is read.
+    Function(fn() -> String),
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/src/defaults.rs:15]: This restriction of `String` only values compared to the typed default values of the existing macro felt like a major loss. I'm assuming this was a conscious sacrifice that was necessary to move to this facet model and there's no way around it.

File: crates/common/tedge_config_engine/src/defaults.rs [hunk 4]
```diff
@@ -0,0 +1,726 @@
...
+        key: Cow<'static, str>,
+        function: DeriveFn,
+    },
+
+    /// A fallback to a key owned by the root config.
+    ///
+    /// Unlike the same-schema fallbacks above, the key always names a key of
+    /// the root config, so mounting a schema under a prefix never rewrites it.
+    FromRoot(&'static str),
+}
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/src/defaults.rs:42]: What about the `default(variable = ...)` variant that is available in the current model? Missed or intentionally skipped?

File: crates/common/tedge_config_engine/macro/tests/schema_features.rs [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    #[test]
+    fn unset_optional_key_reader_field_names_the_source_key() {
+        let mgr = manager();
+        let dto = FeaturesConfigDto::default();
+        let config: FeaturesConfig = mgr.build_reader(&dto, None, "", None).unwrap();
+        assert_eq!(config.c8y.http.key(), "c8y.url");
+    }
+
+    #[test]
+    fn derived_default_follows_a_renamed_source_key() {
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/macro/tests/schema_features.rs:98]: Felt like a misleading name as there is no "renamed" (deprecated aliases) in this test.

File: crates/common/tedge_config_engine/macro/tests/schema_features.rs [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+        }
+    }
+
+    #[test]
+    #[should_panic(expected = "invalid defaults registry")]
+    fn required_fallback_to_a_defaultless_key_is_rejected_at_startup() {
+        ConfigManager::from_schema::<BrokenConfig>(std::path::Path::new("/etc/tedge"));
+    }
+}
+
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/macro/tests/schema_features.rs:156]: Since this test module is primarily checking all the key schema features in their vanilla form, covering the following as well would be nice:
* deprecated_key
* readonly

These are used in other tests but with a more complicated setup. Direct tests here would be better.

File: crates/common/tedge_config_engine/src/defaults.rs [hunk 7]
```diff
@@ -0,0 +1,726 @@
...
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn is_resolvable(&self, key: &str, depth: usize) -> bool {
+        if depth > 10 {
+            return false;
+        }
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/src/defaults.rs]: Though this "cheap loop detection" mechanism works for our config definitions, there is scope for a smarter detection logic.

File: crates/common/tedge_config_engine/src/reader_helpers.rs [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    display_prefix: &str,
+    profile: Option<&str>,
+) -> Result<OptionalConfig<T>, ConfigError>
+where
+    T::Err: std::fmt::Display,
+{
+    let value = match config_get_with_defaults(dto, key, defaults, root_resolver) {
+        Ok(Some(v)) => Some(v),
+        Ok(None) => None,
+        Err(ConfigError::ReflectError(_)) => None,
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/src/reader_helpers.rs]: That `ConfigError::ReflectError(_)` match feels too broad, which might result in even genuine failures like cyclic references to be deemed unset. Might wanna narrow that down to a specific error.

File: crates/common/tedge_config_engine/src/federated.rs [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+        Self {
+            mounts: Vec::new(),
+            config_dir: config_dir.to_owned(),
+        }
+    }
+
+    /// Adds a source under `prefix`; longer prefixes are matched first.
+    ///
+    /// A source using `from_root` defaults can only be mounted once the root
+    /// config (empty prefix) is mounted, and every `from_root` reference is
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/src/federated.rs:35]: Since this `FederatedConfig` type is only relevant when there is a root schema, why not accept it in the constructor itself, so that this case of mounting a schema without its root can be prevented?

File: crates/common/tedge_config_engine/src/defaults.rs [hunk 10]
```diff
@@ -0,0 +1,726 @@
...
+                }),
+                _ => None,
+            })
+            .collect()
+    }
+
+    fn validate(&self) -> Result<(), String> {
+        for field in &self.defaults {
+            if let DefaultSpec::FromKey(source_key) = &field.spec {
+                if !self.is_resolvable(source_key, 0) {
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/src/defaults.rs]: Why restrict the cycle detection to `FromKey` only? It's valid for `FromOptionalKey` and `FromKeyVia` variants as well, right?

**Review by albinsuresh — COMMENTED** [review]
Initial impressions: I really appreciate this facet adoption that resulted in extracting a lot of "magic" from the macro generated code into regular rust code. That change alone eased the reviewing and hopefully maintainability as well. The tests under `tedge_config_engine/macro/tests` and the code documentation really helped with the review. Features like root mounts and external schema definitions are great additions compared to the current humungous monolith config definition.

But what I missed the most is some design doc that captured the following:

* the key motivation behind this migration to facet. Some of it was already documented in the prototype PR. That doc is worth porting to this PR. I was looking for any motivation beyond code simplicity (less magic code) and the sheer volume of the code generated.
* A comparison of what we gained vs lost with facet (e.g: from typed default values in schema definition to string values only)
* While experiencing some of the limitations of the new engine (like the loss of type safety at compile time in some cases), I kept wondering if the reflective data access provided by facet was really worth it. Especially since we don't have the requirement of dynamic schemas (even for custom mappers) that would have required such flexible data access.

Such a doc, esp with that comparison, would have definitely helped.

File: crates/common/tedge_config_engine/macro/tests/schema_features.rs (hunk 6 above)
Comment (albinsuresh) [reply crates/common/tedge_config_engine/macro/tests/schema_features.rs:156]: Ignore that comment. I missed `schema_attrs.rs` which is already covers both.

File: crates/common/tedge_config_engine/src/optional.rs [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+    pub key: String,
+    pub profile: Option<String>,
+}
+
+impl std::fmt::Display for ConfigNotSet {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match &self.profile {
+            Some(profile) => write!(
+                f,
+                "A value for '{key}' is missing (profile '{profile}').\n    \
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/src/optional.rs]:
```suggestion
                "A value for '{key}' (profile '{profile}') is missing.\n    \
```

File: crates/common/tedge_config_engine/src/reader_helpers.rs [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+    defaults: &DefaultsRegistry,
+    root_resolver: RootResolver<'_>,
+    key: &str,
+) -> Result<T, ConfigError>
+where
+    T::Err: std::fmt::Display,
+{
+    let value = config_get_with_defaults(dto, key, defaults, root_resolver)?.ok_or_else(|| {
+        ConfigError::ReflectError(format!(
+            "Required config key '{key}' is not set and has no default"
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/src/reader_helpers.rs]:
```suggestion
            "Required config key '{display_prefix}.{key}' (profile: {profile}) is not set and has no default"
```
To be consistent with error reporting elsewhere.

File: crates/common/tedge_config_engine/src/reader_helpers.rs [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+    };
+
+    let profile_value = profile.map(str::to_owned);
+
+    match value {
+        Some(v) => {
+            let display_key = dotted_key(display_prefix, key);
+            let parsed: T = v.parse().map_err(|e: T::Err| {
+                ConfigError::ParseError(format!(
+                    "Failed to parse value for optional field '{key}': {e}"
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/src/reader_helpers.rs]:
```suggestion
                    "Failed to parse value for optional field '{display_prefix}.{key}' (profile: {profile}): {e}"
```

File: crates/common/tedge_config_engine/src/ops.rs [hunk 14]
```diff
@@ -0,0 +1,145 @@
...
+
+    fn save(&self, dto: &T) -> Result<(), ConfigError> {
+        let content = facet_toml::to_string(dto)
+            .map_err(|e| ConfigError::IoError(format!("serialization error: {e}")))?;
+        if let Some(parent) = self.path.parent() {
+            std::fs::create_dir_all(parent)
+                .map_err(|e| ConfigError::IoError(format!("{}: {e}", parent.display())))?;
+        }
+        std::fs::write(&self.path, content)
+            .map_err(|e| ConfigError::IoError(format!("{}: {e}", self.path.display())))?;
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/src/ops.rs:95]: Wouldn't a temp file write and an atomic move be safer, as done by the current impl?

File: crates/common/tedge_config_engine/src/reflect.rs [hunk 15]
```diff
@@ -0,0 +1,789 @@
...
+            }
+        }
+        (key.to_owned(), None)
+    }
+}
+
+/// Returns an error if the field at `key` is marked read-only via `tedge::readonly`.
+pub fn check_read_only(shape: &'static Shape, key: &str) -> Result<(), ConfigError> {
+    let field = find_leaf_field(shape, key);
+    if let Some(field) = field {
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/src/reflect.rs:120]: Shouldn't this function also do `validate_key` and return an `UnknownKey` error for invalid keys, just like the `config_get`, `config_set` etc does?

File: crates/common/tedge_config_engine/src/reflect.rs [hunk 16]
```diff
@@ -0,0 +1,789 @@
...
+    let new_dto: T = rebuild_dto(dto, key, FieldAction::Set(value))?;
+    *dto = new_dto;
+    Ok(())
+}
+
+/// Rebuilds a DTO with one dotted key reset to its unset `Option` state.
+pub fn config_unset<T: for<'a> Facet<'a>>(dto: &mut T, key: &str) -> Result<(), ConfigError> {
+    validate_key(T::SHAPE, key)?;
+    let new_dto: T = rebuild_dto(dto, key, FieldAction::Unset)?;
+    *dto = new_dto;
```
Comment (albinsuresh) [inline crates/common/tedge_config_engine/src/reflect.rs]: I've been wondering why we need this reflection based approach for `tedge config set/unset`, which is just about mutating the value in a single config file without the burden of cross-file defaults resolutions and all that. I was even wondering if we could use `toml_edit` to do this, once the right file is identified with the prefix. If that's not feasible for reasons I'm overlooking, you could explore the `facet::Poke` type that may help avoid this "rebuild and replace" logic for updating a single value.

File: crates/common/tedge_config_engine/macro/tests/compile_fail/derive_fn_returns_wrong_type.stderr (hunk 2 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/macro/tests/compile_fail/derive_fn_returns_wrong_type.stderr:11]: I don't think so, but trybuild comes with a mode where you set an environment variable and it will update the saved snapshots to match the current behaviour, so the drift is easy to deal with. I can't remember the precise details, but it tells you how to do it in the output of the failing test.

File: crates/common/tedge_config_engine/src/defaults.rs (hunk 3 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/defaults.rs:15]: It's definitely something of a loss, but we can verify defaults work just by creating a config reader somewhere. So essentially as soon as we use the config to its full extent, the default value will be used and we will discover the error. We can also have a test which iterates over the keys and checks the defaults are valid (just like we do already for example values).

File: crates/common/tedge_config_engine/macro/tests/schema_features.rs (hunk 5 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/macro/tests/schema_features.rs:98]: "renamed" in this instance refers to the `rename` attribute (like serde rename, allows the rust name and the config field name to differ) on the source key for the default (which is `device.type` here, the rust name is `ty` to avoid conflicts with the rust `type` keyword).

File: crates/common/tedge_config_engine/src/defaults.rs (hunk 7 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/defaults.rs]: In practice, I expect this to be fine - we might have 2 or 3 layer-deep config key resolution steps, but it shouldn't be more than that. That said, with a more elaborate method that tracks the keys we have consulted, we could keep going until we detect a loop for sure, and provide more information.

I'm also thinking this should be a panic, since it's not something the user has control over.

File: crates/common/tedge_config_engine/src/defaults.rs (hunk 10 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/defaults.rs]: You're right, I will fix that

File: crates/common/tedge_config_engine/src/reflect.rs (hunk 16 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/reflect.rs]: Yeah, I think `facet::Poke` is probably a reasonable way to go, just that the reasonable place to adopt it during the prototype was when I realised the prototype was originally built against a comparatively ancient version of facet, so I was focusing on just getting something working and didn't spot the Poke API at that time.

File: crates/common/tedge_config_engine/src/reader_helpers.rs (hunk 8 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/reader_helpers.rs]: Yeah, I think this needs the error type breaking down as `ReflectError` only has a `String` inside. Alternatively, I should investigate to what extent any error is actually equivalent to `None` here, I think this branch is just wrong, we should be returning the error in every case

File: crates/common/tedge_config_engine/src/defaults.rs (hunk 4 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/defaults.rs:42]: I've now added this. It was an oversight not to include it in the first place, it now exists in the latest commit (59e4362)

File: crates/common/tedge_config_engine/src/federated.rs (hunk 9 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/federated.rs:35]: Perhaps we could support this by checking if any keys within the config reference a root config and add a marker trait depending on whether that is/isn't the case. But I'll need to look into this to establish to what extent this will actually help matters and to what extent it makes the code generation more complicated for little gain.

File: crates/common/tedge_config_engine/src/reader_helpers.rs (hunk 12 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/reader_helpers.rs]: This one along with the other `ReflectError`s really ought to be `panic`s. They're internal and should never be exposed to the user. I will make sure they don't happen with some macro-generated tests.

File: crates/common/tedge_config_engine/src/reflect.rs (hunk 15 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/reflect.rs:120]: Possibly, although I'm not sure it's necessarily worth the complexity of duplicating the validation, particularly given we only call this method just before we validate the key anyway.

File: crates/common/tedge_config_engine/src/reader_helpers.rs (hunk 12 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/reader_helpers.rs]: Addressed by cc47ef49

File: crates/common/tedge_config_engine/src/reader_helpers.rs (hunk 13 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/reader_helpers.rs]: Also removed as part of cc47ef4 as this shouldn't be a user-facing error

File: crates/common/tedge_config_engine/src/reflect.rs (hunk 16 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/reflect.rs]: Addressed by 0ab9109

File: crates/common/tedge_config_engine/src/defaults.rs (hunk 10 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/defaults.rs]: Addressed in 9fa273d1

File: crates/common/tedge_config_engine/src/reader_helpers.rs (hunk 8 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/reader_helpers.rs]: This special error handling was removed in cc47ef49

File: crates/common/tedge_config_engine/src/defaults.rs (hunk 7 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/defaults.rs]: Replaced with proper cycle detection in 9fa273d1

File: crates/common/tedge_config_engine/src/defaults.rs (hunk 3 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/defaults.rs:15]: This auto-generated test is added as part of fdc23820

File: crates/common/tedge_config_engine/src/append_remove.rs (hunk 1 above)
Comment (jarhodes314) [reply crates/common/tedge_config_engine/src/append_remove.rs:72]: They'll be added as needed

File: design/decisions/0012-facet-config.md [hunk 17]
```diff
@@ -0,0 +1,119 @@
+# Facet-powered tedge config support
```
Comment (albinsuresh) [inline design/decisions/0012-facet-config.md:1]: Did you write this doc yourself or has AI writing really improved? I was asking because it was a pleasant read and unlike the recent AI generated docs I didn't have to read each and every sentence multiple times to make sense out if it.

File: design/decisions/0012-facet-config.md [hunk 18]
```diff
@@ -0,0 +1,119 @@
...
+
+### Why facet?
+
+The root of the complexity in the existing macro is that it needs to generate a very large amount of infrastructure for supporting the `tedge config` command. There is something of a direct conflict here between the ability to interact with the configuration by setting a string key to a value to be parsed from a string, and the ability for the code to consume the config as a normal Rust struct.
+
+The macro currently needs to all the heavy lifting as we have no way to introspect the configuration schema at runtime. `facet` is a crate designed to provide a reflection mechanism in Rust. This means we can define a normal Rust struct for the schema, then introspect on it at runtime with normal Rust code. Moving the logic out of the macro will make it much simpler to understand what's going on - "normal" Rust functions are much easier to poke around with, and a simpler macro is easier to understand, e.g. by reading `cargo expand` output.
```
Comment (albinsuresh) [inline design/decisions/0012-facet-config.md]:
```suggestion
The macro currently needs to do all the heavy lifting as we have no way to introspect the configuration schema at runtime. `facet` is a crate designed to provide a reflection mechanism in Rust. This means we can define a normal Rust struct for the schema, then introspect on it at runtime with normal Rust code. Moving the logic out of the macro will make it much simpler to understand what's going on - "normal" Rust functions are much easier to poke around with, and a simpler macro is easier to understand, e.g. by reading `cargo expand` output.
```

File: design/decisions/0012-facet-config.md [hunk 19]
```diff
@@ -0,0 +1,119 @@
...
+
+With the `from_root` attribute, we can validate this when constructing the federated configuration (i.e. the object that contains both the mapper configuration and the core tedge configuration it depends upon). In this example, the fact we are able to do runtime validation is the very reason we _can_ support this federated configuration.
+
+We need to ensure that the macro is adequately tested with negative tests in particular to ensure that invalid configurations are caught without depending on the consumer of the macro manually writing tests.
+
+## Migration structure
+
+The migration to the facet-powered configuration will be a large change, so the implementation will be done in a series of PRs to make it easier to review. These will be focused on providing meaningful externally observable behaviour changes in each PR.
+
+PR 1: Engine crates: The underlying logic for the macro and the `tedge config` command is implemented. This servers as the groundwork for the rest of the change. No changes to the existing `tedge config` behaviour is made in this PR, nor is the new engine wired into any production code paths.
```
Comment (albinsuresh) [inline design/decisions/0012-facet-config.md]:
```suggestion
PR 1: Engine crates: The underlying logic for the macro and the `tedge config` command is implemented. This serves the groundwork for the rest of the change. No changes to the existing `tedge config` behaviour is made in this PR, nor is the new engine wired into any production code paths.
```

**Review by albinsuresh — APPROVED** [review]
LGTM.

File: design/decisions/0012-facet-config.md (hunk 17 above)
Comment (jarhodes314) [reply design/decisions/0012-facet-config.md:1]: Indeed, it was handwritten, with a bit of review (but no actual editing) from claude.

File: design/decisions/0012-facet-config.md (hunk 18 above)
Comment (jarhodes314) [reply design/decisions/0012-facet-config.md]: New addressed

File: design/decisions/0012-facet-config.md (hunk 19 above)
Comment (jarhodes314) [reply design/decisions/0012-facet-config.md]: Now addressed
